### PR TITLE
Improve Emoji Picker Accessibility

### PR DESCRIPTION
**Pull request by slafleche**
This PR aims to improve the accessibility of the emoji picker:

- [x] Focus goes back to toggle button when exiting with escape, or "x"
- [x] If you tab your way out of the toggle button (shift + tab), we close the menu
- [x] If you tab out of the menu (tab from last emoji), the menu closes
- [x] Descriptions and labels were added for clarity
- [x] Added "skip to content" type button to skip past the emoji list to the categories
- [x] Selecting a category will select the first emoji in that category
- [x] Arrow keys work, for now they work as a list, not a grid

Notes:
- I've opted to handle the arrow down as arrow right and arrow up as arrow left. I suspect this is easier to understand for a blind user. Sighted users can use a mouse
- Once you're in the emoji list, there's no easy way to get out except for leaving and coming back. I haven't seen any best practices for this. Maybe one day we'll add a hotkey... but I think this is good enough for now.

Closes: https://github.com/vanilla/vanilla/issues/6728

### CODE REVIEW

File: plugins/rich-editor/src/scripts/components/EditorEmojiButton.jsx [hunk 1]
```diff
@@ -45,13 +46,54 @@ export default class EditorEmojiButton extends React.Component {
         this.props.closeMenu(e);
     }
 
+    /**
+     * Handle key presses
+     * @param {SyntheticEvent} e
```
[the remark this answers is not in the document]
Comment (charrondev) [reply plugins/rich-editor/src/scripts/components/EditorEmojiButton.jsx]: Additionally this should be React.SyntheticEvent. That way you can get editor completion for it.

File: plugins/rich-editor/src/scripts/components/EditorEmojiPicker.jsx [hunk 2]
```diff
@@ -52,6 +56,20 @@ export default class EditorEmojiPicker extends React.Component {
         }
     }
 
+    /**
+     * Close if we lose focus on the component
+     * @param {React.Event} event - A synthetic event.
```
Comment (charrondev) [inline plugins/rich-editor/src/scripts/components/EditorEmojiPicker.jsx]: This should be `React.SyntheticEvent` or better `React.FocusEvent`.

Also the parameter name doesn't line up with the docblock.

**Comment by charrondev** [comment]
A couple additional things:

- You still need to run a build on the rich editor js.
- There are conflicts in the CSS now.

File: plugins/rich-editor/src/scripts/components/EditorEmojiMenu.jsx [hunk 3]
```diff
@@ -113,8 +121,9 @@ export default class EditorEmojiMenu extends React.PureComponent {
         const pos = rowIndex * rowSize + columnIndex;
         const emojiData = emojis[pos];
         let result = null;
+        const selectedButton = this.state.firstEmojiOfGroup >= 0 && this.state.firstEmojiOfGroup === pos ;
```
[the remark this answers is not in the document]
Comment (slafleche) [reply plugins/rich-editor/src/scripts/components/EditorEmojiMenu.jsx]: "is" sounds like a state to me, not an element.

**Pull request merged**